### PR TITLE
MetaDataAPI V2

### PR DESCRIPTION
**Pull request by ArdenHide**

### CODE REVIEW

File: src/MetaDataAPI/Erc20Manager/Erc20Provider.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    {
+        provider = new ERC20CacheProvider();
+    }
+
+    public Erc20Provider(ERC20CacheProvider provider)
+    {
+        this.provider = provider;
+    }
+
+    public Erc20Token GetErc20Token(string rpcUrl, BigInteger chainId, EthereumAddress address)
```
Comment (Lomet) [inline src/MetaDataAPI/Erc20Manager/Erc20Provider.cs:22]: "MAX_SAFE_CHAIN_ID = floor( ( 2**53 - 39 ) / 2 ) = 4503599627370476"
we can use long or long as chainId

**Comment by ArdenHide** [comment]
Let's close issues here:
https://github.com/ArdenHide/NethereumGenerator/issues

Then we can update our package https://github.com/The-Poolz/poolz.finance.csharp.contracts using the newest version of "ArdenHide/NethereumGenerator"

It's need to possible create service which can use `LockDealNFTService` as dependencies and reperesent two functions `FetchPoolInfo` and `IsPoolIdInSupplyRange`.

P.S. Right now it is impossible to create a "correct", testable class. It is possible to use the DI pattern by passing `LockDealNFTService` to a method, but it is preferable to use the class constructor to receive `LockDealNFTService` as a dependency.

P.S.S. The ideal result is that there is no need to create a wrapper class, and we can directly use `ILockDealNFTService` as a dependency.

**Comment by ArdenHide** [comment]
Create saved lambda requests with folow format:

Example of file name: `chainId 56 - poolId 0 - deal provider`
```json
{
  "TestDescription": "Write here what is expected for this request",
  "queryStringParameters": {
    "poolId": 0,
    "chainId": 97
  }
}
```

File: src/MetaDataAPI/Response/ValidationErrorResponse.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
+﻿using System.Net;
+using FluentValidation.Results;
+
+namespace MetaDataAPI.Response;
+
+public class ValidationErrorResponse : LambdaResponse
+{
+    public ValidationErrorResponse(ValidationResult validationResult)
+        : base(
+            body: validationResult.ToString($"{Environment.NewLine}"),
```
Comment (Lomet) [inline src/MetaDataAPI/Response/ValidationErrorResponse.cs:10]: is it mistype?

Comment (ArdenHide) [reply src/MetaDataAPI/Response/ValidationErrorResponse.cs:10]: If you are about "ToString()"

This is a special method of "FluentValidation" library in "ValidationResult" object
image

**Comment by ArdenHide** [comment]
New format of description:
```
ChainId: 4503599627370476, PoolId: 4503599627370476, Hash: HASH_OF_ALL_PARAMS_HERE
```

**Pull request merged**